pymc/distributions/distribution.py

```diff
@@ -622,13 +679,16 @@ def dist(
         if logcdf is None:
             logcdf = default_not_implemented(class_name, "logcdf")
 
+        def dist_moment(rv, size, *dist_params):
+            # TODO: add check for other op with inner graph (not Scan)
+            fgraph = FunctionGraph(outputs=[dist(*dist_params, size=size)], clone=True)
```

Comment:
@ricardoV94 do we need some check here for` OpFromGraph`, `Composite`, and `ScalarLoop` Ops in `fgraph`? I mean to use default moment like we did before in this case.

Reply:
Yes, let's raise `NotImplementedError` for those, and default back to the zeros like before. I think we should put this `dist_moment` helper inside the old `default_moment` function?

Reply:
Wait only `OpFromGraph` is problematic. The others can't have `RandomVariables` inside so we can just replace their inputs outside without concern for their inner graph.

Reply:
Then we can add a github issue to also implemnet the OpFromGraph case which shouldn't be too hard (similar to scan). Or do you want to do that in this PR after all? That would neatly conclude all cases.

Reply:
@ricardoV94 Well, I think I want to implement  the OpFromGraph case in this PR. Do you now some example on how to use OpFromGraph? I need to create some test case for it

Reply:
I'll write something tomorrow. Thanks for the initiative in advance!